extractor.go

```diff
@@ -80,100 +189,116 @@ func (te *Extractor) Extract(reader io.Reader) error {
 	return nil
 }
 
-// Sanitize sets up the extractor to use built in sanitation functions
-// (Modify paths to be platform legal, symlinks may not escape extraction root)
-// or unsets any previously set sanitation functions on the extractor
-// (no special rules are applied when extracting)
-func (te *Extractor) Sanitize(toggle bool) {
-	if toggle {
-		te.SanitizePathFunc = sanitizePath
-		te.LinkFunc = func(inLink Link) error {
-			if err := childrenOnly(inLink); err != nil {
-				return err
-			}
-			if err := platformLink(inLink); err != nil {
-				return err
-			}
-			return os.Symlink(inLink.Target, inLink.Name)
-		}
-	} else {
-		te.SanitizePathFunc = nil
-		te.LinkFunc = nil
+// validateTarPath returns an error if the path has problematic characters
+func validateTarPath(tarPath string) error {
+	if len(tarPath) == 0 {
+		return fmt.Errorf("path is empty")
+	}
+
+	if tarPath[0] == '/' {
+		return fmt.Errorf("%q : path starts with '/'", tarPath)
 	}
-}
 
-// outputPath returns the path at which to place tarPath
-func (te *Extractor) outputPath(tarPath string) (outPath string, err error) {
 	elems := strings.Split(tarPath, "/") // break into elems
 	for _, e := range elems {
-		if e == ".." {
-			return "", fmt.Errorf("%s : path contains '..'", tarPath)
+		switch e {
+		case "", ".", "..":
+			return fmt.Errorf("%q : path contains %q", tarPath, e)
 		}
 	}
-	elems = elems[1:]                       // remove original root
-	outPath = strings.Join(elems, "/")      // join elems
-	outPath = gopath.Join(te.Path, outPath) // rebase on to extraction target root
-	// sanitize path to be platform legal
-	if te.SanitizePathFunc != nil {
-		outPath, err = te.SanitizePathFunc(outPath)
-	} else {
-		outPath = fp.FromSlash(outPath)
-	}
-	return
+	return nil
 }
 
-func (te *Extractor) extractDir(h *tar.Header, depth int) error {
-	path, err := te.outputPath(h.Name)
-	if err != nil {
-		return err
+// getRelativePath returns the relative path between rootTarPath and tarPath. Assumes both paths have been cleaned.
+// Will error if the tarPath is not below the rootTarPath.
+func getRelativePath(rootName, tarPath string) (string, error) {
+	pathElems := strings.Split(tarPath, "/")
+
+	if len(pathElems) <= 1 {
```

Comment:
```suggestion
	if !strings.ContainsAny(tarPath, "/") {
		return "", errInvalidRootMultipleRoots
	}
```

Reply:
this whole check seems unnecessary, was removed